src/scene/mesh.js

```diff
@@ -630,6 +629,65 @@ Object.assign(pc, function () {
                 // remove data
                 this._geometryData.indices = null;
             }
+        },
+
+        generateWireframe: function () {
+            var typedArray = function (indexBuffer) {
+                switch (indexBuffer.format) {
+                    case pc.INDEXFORMAT_UINT8:
+                        return new Uint8Array(indexBuffer.storage);
+                    case pc.INDEXFORMAT_UINT16:
+                        return new Uint16Array(indexBuffer.storage);
+                    case pc.INDEXFORMAT_UINT32:
+                        return new Uint32Array(indexBuffer.storage);
+                    default:
+                        return null;
+                }
+            };
+
+            var lines = [];
+            var format;
+            if (this.indexBuffer[0]) {

```

Comment:
maybe should be if (this.indexBuffer && this.indexBuffer[0]) to handle case when indexBuffer in undefined

Reply:
this.indexBuffer is initialized to [null] at construction and (far as I know) should always be an array.

Adding the test could be safer, but it also implies to the reader that indexBuffer could validly be undefined or null. Is that the case?

Reply:
According to the doc the array could be null, so perhaps the test should be:
```this.indexBuffer.length > 0 && indexBuffer[0]``` ?

Reply:
it's public variable, so the user could set it to just null.
I don't remember specific test case, but I had this issue with my code where indexBuffer was null, so it's safer to test it.

Reply:
I would argue that's a user error. What if a user sets indexBuffer to a string or map or anything else? We can't test all variables all the time and I think the code should reflect what the system considers to be valid state.

Reply:
And the valid state is defined by the property doc:
```
     * @property {pc.IndexBuffer[]} indexBuffer An array of index buffers. For unindexed meshes, this array can
     * be empty. The first index buffer in the array is used by {@link pc.MeshInstance}s with a renderStyle
     * property set to pc.RENDERSTYLE_SOLID. The second index buffer in the array is used if renderStyle is
     * set to pc.RENDERSTYLE_WIREFRAME.
```

Reply:
fair enough.

Reply:
I actually think adding a check for null would do a disservice, because then a reader of the code would think that null is a valid state.

Reply:
I guess this is where typescript would help us; the setter function would have a signature disallowing any user from setting indexbuffer to null.

Reply:
Additional array check is being added in https://github.com/playcanvas/engine/pull/2024